src/Caching/Storages/FileStorage.php

```diff
@@ -143,7 +143,7 @@ public function lock($key)
 	{
 		$cacheFile = $this->getCacheFile($key);
 		if ($this->useDirs && !is_dir($dir = dirname($cacheFile))) {
-			@mkdir($dir); // @ - directory may already exist
+			@mkdir($dir, 0777, true); // @ - directory may already exist
```

Comment:
@dg Imho, purpose is when `$cache->derive('a')->derive('b')->derive('c')->save('key', 'val');` will create `tmp/_a/_b/_c`, now it creates `tmp/_%00a%00b%00c`.